### ISSUE TITLE
Unable to create account or reset password

### ISSUE THREAD

**Issue by dehboxturtle** [aws-amplify/amplify-ui-android#38]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui-android/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui-android/blob/main/CONTRIBUTING.md#reporting-bugsfeature-requests).

### Which UI component?

Authenticator

### Gradle script dependencies

```groovy
plugins {
    id 'com.android.application'
    id 'org.jetbrains.kotlin.android'
    id 'kotlin-kapt'
    id 'com.google.dagger.hilt.android'
}

android {
    namespace 'com.android.dehboxturtle.playmate'
    compileSdk 33

    defaultConfig {
        applicationId "com.android.dehboxturtle.playmate"
        minSdk 24
        targetSdk 33
        versionCode 1
        versionName "0.0.1"

        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
        vectorDrawables {
            useSupportLibrary true
        }
    }

    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
        }
    }
    compileOptions {
        coreLibraryDesugaringEnabled true
        sourceCompatibility JavaVersion.VERSION_17
        targetCompatibility JavaVersion.VERSION_17
    }
    kotlinOptions {
        jvmTarget = '17'
    }
    buildFeatures {
        compose true
    }
    composeOptions {
        kotlinCompilerExtensionVersion '1.4.7'
    }
    packagingOptions {
        resources {
            excludes += '/META-INF/{AL2.0,LGPL2.1}'
        }
    }
    buildToolsVersion '31.0.0'
}

dependencies {
    kapt 'com.google.dagger:hilt-compiler:2.46.1'
    coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:2.0.3'

    // Amplify core dependency
    implementation 'com.amplifyframework:core-kotlin:2.8.4'
    implementation 'com.amplifyframework:aws-api:2.8.4'
    implementation 'com.amplifyframework:aws-auth-cognito:2.8.4'
    implementation 'com.amplifyframework:aws-api:2.8.4'
    implementation 'com.amplifyframework:aws-storage-s3:2.8.4'
    implementation 'com.amplifyframework.ui:authenticator:1.0.0-dev-preview.1'

    implementation 'com.google.dagger:hilt-android:2.46.1'
    implementation 'androidx.core:core-ktx:1.10.1'
    implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.6.1'
    implementation 'androidx.activity:activity-compose:1.7.1'
    implementation platform('androidx.compose:compose-bom:2023.05.01')
    implementation 'androidx.compose.ui:ui'
    implementation 'androidx.compose.ui:ui-graphics'
    implementation 'androidx.compose.ui:ui-tooling-preview'
    implementation 'androidx.compose.material3:material3'
    implementation 'androidx.compose.material:material-icons-core'
    implementation 'androidx.activity:activity-compose:1.7.1'
    implementation 'androidx.lifecycle:lifecycle-viewmodel-compose:2.6.1'
    testImplementation 'junit:junit:4.13.2'
    androidTestImplementation 'androidx.test.ext:junit:1.1.5'
    androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.1'
    androidTestImplementation platform('androidx.compose:compose-bom:2023.05.01')
    androidTestImplementation 'androidx.compose.ui:ui-test-junit4'
    debugImplementation 'androidx.compose.ui:ui-tooling'
    debugImplementation 'androidx.compose.ui:ui-test-manifest'
}

// Allow references to generated code
kapt {
    correctErrorTypes true
}

```

### Environment information

```
------------------------------------------------------------         
Gradle 8.0                                                           
------------------------------------------------------------         
                                                                     
Build time:   2023-02-13 13:15:21 UTC                                
Revision:     62ab9b7c7f884426cf79fbedcf07658b2dbe9e97               
                                                                     
Kotlin:       1.8.10                                                 
Groovy:       3.0.13                                                 
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          17.0.7 (Amazon.com Inc. 17.0.7+7-LTS)
OS:           Windows 10 10.0 amd64



```

### Please include any relevant guides or documentation you're referencing

https://ui.docs.amplify.aws/android/connected-components/authenticator

### Describe the bug

When using the authenticator UI I tried to create a user and it shows a generic error when you fill out the form. 
![image](https://github.com/aws-amplify/amplify-ui-android/assets/58672268/80047ca4-f6fd-44f6-87e8-53b9079ee2f1)

If I create the user manually through the amplify UI it lets me log in but it forces you to change the temporary password which fails with a generic error as well. Also the snackbar for the error isn't pinned to the bottom of the page.
![image](https://github.com/aws-amplify/amplify-ui-android/assets/58672268/d32a82b6-5d59-468f-9871-6eccb649aa37)

### Reproduction steps (if applicable)

1. Attempt to create user in amplify UI
2. observe error

1. Attempt to reset password for user create in amplify web console
2. observer error

### Code Snippet

```kotlin
override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            Authenticator { state ->
                PlayMateTheme {
                    // A surface container using the 'background' color from the theme
                    Surface(
                        modifier = Modifier.fillMaxSize(),
                        color = MaterialTheme.colorScheme.background
                    ) {
                        Greeting("PlayMate")
                    }
                }
            }
        }
...
}

```

### Log output

No log info came out with the failure unfortunately.

**Comment by mattcreaser** [comment]
Hi @dehboxturtle.

The snackbar appears at the bottom of the Authenticator by default, so you can make it appear at the bottom of the screen by making the Authenticator full size:

```kotlin
Authenticator(
    modifier = Modifier.fillMaxSize()
) {
  // ...
}
```

Alternatively, you could instead use `onDisplayMessage` to display the messages wherever you want.

```kotlin
Authenticator(
    onDisplayMessage = { /* ... */ }
) {
  // ...
}
```

For tracking down the error can you please add the logging plugin to Amplify in your `Application` class (add it as the first plugin applied), and then capture the logs? That should help with some more information about the source of the error.

```kotlin
Amplify.addPlugin(AndroidLoggingPlugin(LogLevel.VERBOSE))
Amplify.addPlugin(AWSCognitoAuthPlugin())
Amplify.configure(this)
```

**Comment by mattcreaser** [comment]
We've also released `1.0.0-dev-preview.2` that has an important additional log statement for getting information about errors after adding the `AndroidLoggingPlugin` as above. The logged warning should have detailed information about the source of the error.

**Comment by dehboxturtle** [comment]
Hello! Thanks for the tips :) After investigating a bit and trying your recommendations I was able to resolve it by removing the fields that were not in the authenticator. I believe it was probably the profile picture attribute. 

I was able to create a user normally via the app at that point. Is there somewhere in the docs that the supported attributes are listed? I'm just curious if I missed some documentation when I was setting up the Auth flow.
